src/env.types.ts

```diff
@@ -0,0 +1,47 @@
+export type Env = GitLabCIPredefinedVariables &
+  MergeRequestVariables & {
+    GITLAB_HOST: string
+
+    GITLAB_TOKEN: string
+    GITLAB_TOKEN_TYPE: LooseString<'job' | 'oauth'>
+    GITLAB_CI_USER_NAME: string | undefined
+    GITLAB_CI_USER_EMAIL: string
+    GITLAB_COMMENT_TYPE: LooseString<'discussion' | 'note'>
+    DEBUG_GITLAB_CREDENTIAL: LooseString<'1' | 'true'>
+
+    HOME: string
+    NPM_TOKEN: string | undefined
+  }
+
+type MergeRequestVariables =
+  | {
+      // this is used to be checked if the current job is a merge request job
+      CI_MERGE_REQUEST_SOURCE_BRANCH_NAME: undefined
```

Comment:
```suggestion
      CI_MERGE_REQUEST_SOURCE_BRANCH_NAME?: void
```

or `never`? Ideally, should use `'CI_MERGE_REQUEST_SOURCE_BRANCH_NAME' in env` check to split the two types. But if cannot be done easily, it's fine to keep as-is.

Reply:
Weird, it seems that only `undefined` works. I'll keep it like this for now.

Reply:
OK